File: Utils/src/hkccpacmanrobot/utils/map/MapUnit.scala
package hkccpacmanrobot.utils.map

/**
 * Created by beenotung on 3/21/15.
 */
abstract class MapUnit (val key: MapKey,val  value: MapContent=new MapContent) extends Cloneable{

  abstract  def getLatestTime: Long


  def setTime(Time: Long, preservedLong: Long)
}

File: Utils/src/hkccpacmanrobot/utils/map/MapUnit.java
package hkccpacmanrobot.utils.map;

public abstract class MapUnit implements Cloneable {
    public final MapKey key;
    public final MapContent value;


    public MapUnit(MapKey key) {
        this.key = key;
        this.value = new MapContent();
    }

    public MapUnit(MapKey key, MapContent value) {
        this.key = key;
        this.value = value;
    }

    public abstract long getLatestTime();

    public abstract void setTime(long Time, long preservedLong);
}
